File: clients/JavaScript/model/action.js
const EntityAction = require('./entity-action');
class Action {
    constructor(entityActions) {
        this.entityActions = entityActions;
    }
    static async readFrom(stream) {
        let entityActions;
        entityActions = new Map();
        for (let i = await stream.readInt(); i > 0; i--) {
            let entityActionsKey;
            let entityActionsValue;
            entityActionsKey = await stream.readInt();
            entityActionsValue = await EntityAction.readFrom(stream);
            entityActions.set(entityActionsKey, entityActionsValue);
        }
        return new Action(entityActions);
    }
    async writeTo(stream) {
        let entityActions = this.entityActions;
        await stream.writeInt(entityActions.size);
        for (let entityActionsEntry of entityActions) {
            let entityActionsKey = entityActionsEntry[0];
            let entityActionsValue = entityActionsEntry[1];
            await stream.writeInt(entityActionsKey);
            await entityActionsValue.writeTo(stream);
        }
    }
}
module.exports = Action

File: clients/Ruby/model/action.rb
require_relative 'entity_action'
class Action
    attr_accessor :entity_actions
    def initialize(entity_actions)
        @entity_actions = entity_actions
    end
    def self.read_from(stream)
        entity_actions = Hash.new
        stream.read_int().times do |_|
            entity_actions_key = stream.read_int()
            entity_actions_value = EntityAction.read_from(stream)
            entity_actions[entity_actions_key] = entity_actions_value
        end
        Action.new(entity_actions)
    end
    def write_to(stream)
        stream.write_int(@entity_actions.length())
        @entity_actions.each do |key, value|
            stream.write_int(key)
            value.write_to(stream)
        end
    end
end
